src/es/components/controllers/events/Events.js

```diff
@@ -69,50 +97,79 @@ export default class Events extends Shadow() {
       ...options
     }, ...args)
 
-    this.resetButton = this.root.querySelector('.filter-button-hidden');
+    /** 
+     * calculated dates between min/max actual date expl.: 11.10.2023+—+26.10.2023 would be 11.10.2023 + 12.10.2023 + 13.10.20203 + ... + 26.10.2023
+     * @type {Map<string, Date[]>}
+     */
+    this.dateArrayMap = new Map()
+    // gets fed by get-date-option from the flatpickr, this here is just a default value
+    this.dateStrSeparator = ' — '
 
+    // dispatches 'list-events'
     this.requestListEventsListener = event => {
       this.dispatchEvent(new CustomEvent('list-events', {
         detail: {
-          fetch: this.fetch.then(result => ({
-            events: event.detail
-              ? result.events.filter(resultEvent => {
-                const filterType = event.detail?.this.getAttribute('filter-type')
-                // show reset button
-                this.resetButton.classList.remove('filter-button-hidden');
-                // TODO: match Flatpickr tags: ["2023-10-10"] with eventDate:("24.4.2024 19:30:0")
-                //if (filterType === 'eventDate' && typeof event.detail.tags[0] ===  'string') event.detail.tags[0] = new Date(...(event.detail.tags[0].replace(/\s.*/, '').split('.').reverse().map((date, i) => i === 1 ? Number(date)-1 : date)))
-                return resultEvent?.[filterType] === event.detail.tags[0]
-              })
-              : result.events,
-            translations: result.translations
-          }))
+          fetch: this.fetch.then(result => {
+            let events
+            if (event && event.detail && event.detail.this && event.detail.this.getAttribute('filter-type') && event.detail.tags && event.detail.tags[0]) {
```

Comment:
I am not sure, but this line does the same like:
event?.details?.this?.getAttribute('filter-type') && event?.detail?.tags?[0]   ?

Reply:
If yes, we can simply it here and some other places too :)